### DISCUSSION TITLE
GDP for each settlement

### DISCUSSION THREAD

**Discussion by mokun** [mars-sim/mars-sim#477] [Ideas]
It would be beneficial to calculate the GDP for each settlement as a kind of benchmark for production of goods and services.

Of course, it begs the definition of what GDP is and what it represents. 

The real world GDP is defined like [this](https://www.investopedia.com/articles/investing/051415/how-calculate-gdp-country.asp).

But in case of mars-sim, we'll need to roll in our own in-house definition on what it comprises.

@bevans2000 @Urwumpe 
Any thought on exactly what should be counted toward its calculation and what implementation approach to undertake ?

**Comment by Urwumpe** [comment]
I would not recommend to use GDP or similar there, because it might be the wrong metric in the Mars colony scenario right now. Because of the situation in which we are mostly simulating, a base with a high GDP might be economically successful, but won't survive for long. Also, research is really hard to include there and we have a lot of research initially. (We can only measure how much we invest into science, not how much we gain by it - a country gaining a lot of knowlegde is only successful in GDP, if the science can be applied to something economic. Medicine can even have a negative effect on GDP: If you need to spend less money for staying healthy, the GDP is lower - a health system like the US with its excessive costs is favorable in the GDP). It makes more sense to use, when the colonists reach the top of the Maslow pyramid. So, if you want to use it as secondary measurement in more complex scenarios, when other metrics between the bases are too similar, thats fine for me. 

Just like many people slowly realize on Earth, survival is not mandatory anymore.

From the common alternatives to purely looking at the GDP, I would recommend taking a look at the Human Development Index of the United Nations or the Better Life Index of the OECD. The Green GDP might be solution closer to the GDP, but again, will likely not fit the bill of a Mars colony.

I would recommend using the BLI as template and adapt it to Mars (without the vagueness of the real one) for our usual small Mars population, so we have again about 11 dimensions to compare in detail. Important note in our case: We have TWO environments to consider. The external Environment is Mars. I think, from an economic POV, this metric would have to measure how close we are to a pure, untouched Mars environment. But we also have an internal environment, the habitat of the humans. Right now it all happens inside bases, Terraforming might grow it larger for all - if we have millions of CPU-hours. We need to measure how robust and stable this environment is: How much oxygen does the base produce? How much water? How many systems can fail before the environment will collapse in a chain reaction? In our case, this is really important and can be quantified, since we know our simulation. 

The metric goal would be living like in a developed country on Earth - but on Mars. By having more dimensions to look at in detail, we can maybe even let the AI Settlers themselves decide about the strategy to reach full development. And maybe have social scientists produce reports about the problems that make them miss benchmarks and goals.

**Reply by mokun** [reply]
>  the Human Development Index of the United Nations or the Better Life Index of the OECD.

Sure. Better Life Index (BLI) sounds cool.

We already have some category scores set up to rank each settlement in the console. 

But we can beef them up and have the settlers do actually work to collect these data.

And we may as well make these indices or ranking available in Command Dashboard for easy viewing and comparison.
